File: src/main/java/ru/nucodelabs/gem/view/DialogsModule.java
package ru.nucodelabs.gem.view;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.name.Named;
import javafx.scene.control.ButtonType;
import javafx.scene.control.Dialog;

public class DialogsModule extends AbstractModule {
    @Provides
    @Named("Save")
    private Dialog<ButtonType> provideSaveDialog(@Named("CSS") String stylesheet) {
        Dialog<ButtonType> saveDialog = new Dialog<>();
        saveDialog.setTitle("Сохранение");
        saveDialog.setContentText("Сохранить изменения?");
        saveDialog.getDialogPane().getButtonTypes()
                .addAll(ButtonType.YES,
                        ButtonType.NO,
                        ButtonType.CANCEL);
        saveDialog.getDialogPane().getStylesheets().add(stylesheet);
        return saveDialog;
    }
}

File: src/main/java/ru/nucodelabs/gem/view/DialogsModule.kt
package ru.nucodelabs.gem.view

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.google.inject.name.Named
import javafx.scene.control.ButtonType
import javafx.scene.control.Dialog

class DialogsModule : AbstractModule() {
    @Provides
    @Named("Save")
    fun provideSaveDialog(@Named("CSS") stylesheet: String): Dialog<ButtonType> =
        Dialog<ButtonType>().apply {
            title = "Сохранение"
            contentText = "Сохранить изменения?"
            dialogPane.buttonTypes.addAll(ButtonType.YES, ButtonType.NO, ButtonType.CANCEL)
            dialogPane.stylesheets.add(stylesheet)
        }
}
